tests/integration/update.bats

```diff
@@ -584,10 +584,52 @@ EOF
 	check_cgroup_value "cpu.rt_runtime_us" 600001
 }
 
-@test "update devices [minimal transition rules]" {
+@test "update devices limits access [minimal transition rules]" {
 	[[ "$ROOTLESS" -ne 0 ]] && requires rootless_cgroup
-	# This test currently only makes sense on cgroupv1.
-	requires cgroups_v1
+
+	requires root
+
+	# Run a basic shell script that tries to read from to /dev/private,
+	# a dummy device it should not have access to. During "runc update",
+	# it should still not be able to read from the file/device.
+	update_config '.process.args |= ["sh", "-c", "while true; do head -c 10 /dev/private 2>&1 | grep -v \"head: /dev/private: Operation not permitted\"; done"]'
```

Comment:
TODO: Switch to /dev/kmsg instead. Didn't do that initially due to permission issues; see https://github.com/opencontainers/runc/pull/3069.